### ISSUE TITLE
Support communications per resource group

### ISSUE THREAD

**Issue by dsaydon90** [kubeshop/botkube#508]
**Is your feature request related to a problem? Please describe.**
In my case, I want to send the default notifications to slack channel `botkube-all` and I want to send only pod `error` events to a different slack channel `botkube-pod-errors`

**Describe the solution you'd like**

**option 1** 
If the resources could be grouped by a group name. For example
```
config:
  ## Resources you want to watch
  resources:
    - default_group:
      - name: v1/pods 
        namespaces: 
          include:
            - all
          ignore:  
            -   
        events:  
          - create
          - delete
          - error
      - name: v1/services
        namespaces:
          include:
            - all
          ignore:
            -
        events:
          - create
          - delete
          - error
    - group2:
      - name: v1/pods
        namespaces:
          include:
            - all
          ignore:
            - 
        events:
          - error
communications:
  slack:
    enabled: true
    token: 'SLACK_API_TOKEN'
    channel: 'default channel'
    channels:
      defaut_group:
        channel: 'SLACK_CHANNEL'
      group2:
        channel: 'SLACK_CHANNEL'
``` 

**option 2**
Add `tags` (or something else) to each resource, which will help to override some configuration. For Example:
note: for this need additional key like `resource_name`
```
config:
  ## Resources you want to watch
  resources:
    - resource_name: pods
      name: v1/pods 
      namespaces: 
        include:
          - all
        ignore: 
          - 
      events:
        - create
        - delete
        - error
    - resource_name: services
      name: v1/services
      namespaces:
        include:
          - all
        ignore:
          -
      events:
        - create
        - delete
        - error
    - resource_name: only-pod-errors
      name: v1/pods 
      namespaces: 
        include:
          - all
        ignore: 
          - 
      events:
        - error        
      tags:
        communications:
          - slack_channel: 'botkube-pod-errors' # override chaneel        
```

**Describe alternatives you've considered**
Currently, I can think only about duplicating the `botkube` to 2 different releases with different config, but I reallt want to avoid that.

**Comment by PrasadG193** [comment]
I think profiling would be a better approach. And we can bind multiple profiles to different channels. Something we tried to solve with - https://github.com/infracloudio/botkube/pull/291. It became a bit complicated as we tried to address each component - resource, event types, namespace, kubectl verbs. We can try to limit the config and have a cleaner solution. Which should be easier to cofigure for users.
